pkg/conversion/mapi2capi/powervs.go

```diff
@@ -0,0 +1,307 @@
+
+	// Override the reference so that it matches the AWSMachineTemplate.
+	powerVSMachineSet.Spec.Template.Spec.InfrastructureRef.Kind = ibmPowerVSTemplateKind
+	powerVSMachineSet.Spec.Template.Spec.InfrastructureRef.Name = powerVSMachineTemplate.Name
+
+	if m.infrastructure == nil || m.infrastructure.Status.InfrastructureName == "" {
+		errs = append(errs, field.Invalid(field.NewPath("infrastructure", "status", "infrastructureName"), m.infrastructure.Status.InfrastructureName, "infrastructure cannot be nil and infrastructure.Status.InfrastructureName cannot be empty"))
+	} else {
+		powerVSMachineSet.Spec.Template.Spec.ClusterName = m.infrastructure.Status.InfrastructureName
+		powerVSMachineSet.Spec.ClusterName = m.infrastructure.Status.InfrastructureName
+	}
+
+	if len(errs) > 0 {
+		return nil, nil, warnings, utilerrors.NewAggregate(errs)
+	}
+
+	return powerVSMachineSet, powerVSMachineTemplate, warnings, nil
+}
+
+// powerVSProviderSpecFromRawExtension unmarshalls a raw extension into an PowerVSMachineProviderSpec type.
+func powerVSProviderSpecFromRawExtension(rawExtension *runtime.RawExtension) (mapiv1.PowerVSMachineProviderConfig, error) {
+	if rawExtension == nil {
+		return mapiv1.PowerVSMachineProviderConfig{}, nil
+	}
+
+	spec := mapiv1.PowerVSMachineProviderConfig{}
+	if err := yaml.Unmarshal(rawExtension.Raw, &spec); err != nil {
+		return mapiv1.PowerVSMachineProviderConfig{}, fmt.Errorf("error unmarshalling providerSpec: %w", err)
+	}
+
+	return spec, nil
+}
+
+// toPowerVSMachine converts PowerVSMachineProviderConfig to IBMPowerVSMachineSpec.
+func (m *powerVSMachineAndInfra) toPowerVSMachine(providerSpec mapiv1.PowerVSMachineProviderConfig) (*capibmv1.IBMPowerVSMachine, field.ErrorList) {
+	fldPath := field.NewPath("spec", "providerSpec", "value")
+
+	var errs field.ErrorList
+
+	serviceInstance, err := convertServiceInstanceToCAPI(fldPath.Child("serviceInstance"), providerSpec.ServiceInstance)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	image, err := convertImageToCAPI(fldPath.Child("image"), providerSpec.Image)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	network, err := convertNetworkToCAPI(fldPath.Child("network"), providerSpec.Network)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	spec := capibmv1.IBMPowerVSMachineSpec{
+		//	ServiceInstanceID: Deprecated, Use ServiceInstance.
+		ServiceInstance: serviceInstance,
+		SSHKey:          providerSpec.KeyPairName,
+		Image:           image,
+		// ImageRef: Not required as image is set above
+		SystemType:    providerSpec.SystemType,
+		ProcessorType: capibmv1.PowerVSProcessorType(providerSpec.ProcessorType),
+		Processors:    providerSpec.Processors,
+		MemoryGiB:     providerSpec.MemoryGiB,
+		Network:       network,
+		// ProviderID. This is populated when this is called in higher level funcs (ToMachine(), ToMachineSet()).
+	}
+
```

Comment:
Are there any fields from the providerSpec that aren't consumed in this conversion?